### PR TITLE
OO principles

### PR DESCRIPTION
**Pull request by Hartwelg**

### CODE REVIEW

File: src/main/java/cs361/battleships/models/Battleship.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+public class Battleship extends Ship {
+
+    private Square captainquarters;
+
+
+    public Battleship (){
+        super();
+        super.setLength(4);
+        this.captainquarters = new Square();
+        super.setFloatstatus(true);
```
Comment (Crollt) [inline src/main/java/cs361/battleships/models/Battleship.java]: What is float status for?

File: src/main/java/cs361/battleships/models/Ship.java [hunk 2]
```diff
@@ -21,16 +25,25 @@ public Ship(String kind) {
 		switch (kind) {
```
Comment (Crollt) [inline src/main/java/cs361/battleships/models/Ship.java]: Do we still need this switch?

File: src/main/java/controllers/ApplicationController.java [hunk 3]
```diff
@@ -22,6 +21,19 @@ public Result newGame() {
     public Result placeShip(Context context, PlacementGameAction g) {
         Game game = g.getGame();
         Ship ship = new Ship(g.getShipType());
```
Comment (Crollt) [inline src/main/java/controllers/ApplicationController.java]: Don't need this line if we immediately reassign ship below.

File: src/main/java/cs361/battleships/models/Ship.java (hunk 2 above)
Comment (Flazzing) [reply src/main/java/cs361/battleships/models/Ship.java]: I think it's fine if we leave it. No need to remove it, if it doesn't affect the program.

File: src/main/java/cs361/battleships/models/Battleship.java (hunk 1 above)
Comment (Flazzing) [reply src/main/java/cs361/battleships/models/Battleship.java]: To check if the ship is float or sunk. Requested at #38 by @Crollt

File: src/main/java/controllers/ApplicationController.java (hunk 3 above)
Comment (Flazzing) [reply src/main/java/controllers/ApplicationController.java]: Thank for the alignment. Appreciate it

File: src/main/java/cs361/battleships/models/Battleship.java (hunk 1 above)
Comment (Crollt) [reply src/main/java/cs361/battleships/models/Battleship.java]: Okay.

Comment (Bizjakj) [reply src/main/java/cs361/battleships/models/Battleship.java]: Does this do anything  in our current implementation?

**Pull request merged**